libs/deal-ticket/src/components/deal-ticket/deal-ticket.tsx

```diff
@@ -465,7 +467,18 @@ export const DealTicket = ({
           <TimeInForceSelector
             value={field.value}
             orderType={type}
-            onSelect={field.onChange}
+            onSelect={(value) => {
+              if (
+                iceberg &&
+                [
```

Comment:
**chore (blocking)**: this check is repeated three times in the cdiff, so it is probably worth refactoring to its own shared function that expresses the _why_ of the check a bit more. Both IOC and FOK are non-persistent, which is why they can't be icebergs, so perhaps `isNonPersistentOrder` (or `isPersistentOrder` that returns the inverse, obviously).

Otherwise the change looks good

Reply:
@edd @bglownia I'm going to make this change so I can get this in before deploying fairground